### PR TITLE
SERVER-1077: s3 backup and restore preview

### PR DESCRIPTION
**Pull request by annabaker**
# Description
Polishing of #5374 + preface branch with 'server/' for pdf gen, and attaching '-preview' to generate preview site.

# Reasons
https://circleci.atlassian.net/browse/SERVER-1077

### CODE REVIEW

File: jekyll/_cci2/server-3-operator-backup-and-restore.adoc [hunk 1]
```diff
@@ -366,6 +370,121 @@ velero-68788b675c-dm2s7   1/1     Running   0          2m
...
+=== Step 1 - Configure `mc` client
+
+To start, configure https://docs.min.io/minio/baremetal/reference/minio-cli/minio-mc.html[`mc`] to connect to your storage
+provider:
+
+[source,bash]
+----
+# Alias can be any name as long as you use the same value in subsequent commands
+export ALIAS=my-provider
+// Can we clarify what these 3 params are exactly
```
[the remark this answers is not in the document]
Comment (nsmith5) [reply jekyll/_cci2/server-3-operator-backup-and-restore.adoc:394]: Ah right so these are all attributes of the object storage provider Minio. So we could say perhaps:
<YOUR_MINIO_ENDPOINT>, <YOUR_MINIO_ACCESS_KEY_ID> <YOUR_MINIO_SECRET_ACCESS_KEY>? Does that clarify?

I think one aspect that can be confusing is that Minio supports the same API as AWS S3 so there are shared terms there (access key, secret key etc), but its not related to AWS other than that. Velero simply uses this storage for backups but in this section we're not configuring Velero _yet_. Just setting up storage in Minio.

File: jekyll/_cci2/server-3-operator-backup-and-restore.adoc [hunk 2]
```diff
@@ -366,6 +370,121 @@ velero-68788b675c-dm2s7   1/1     Running   0          2m
...
+----
+
+=== Set 3 - Create a user and policy
+
+Next, create a user and policy for Velero to access your bucket.
+
+[source, bash]
+----
+# Create user
+mc admin user add $ALIAS <YOUR_VELERO_ACCESS_KEY_ID> <YOUR_VELERO_SECRET_ACCESS_KEY>
```
Comment (nsmith5) [inline jekyll/_cci2/server-3-operator-backup-and-restore.adoc:416]: Ah, I see why this is a little confusing. We state <YOUR_VELERO_ACCESS_KEY_ID> here. The intention is that is the the key id for Velero to access Minio, but it reads like the _your_ key to access Velero. @annabaker do you have any suggestions on a more clear variable name?

File: jekyll/_cci2/server-3-operator-backup-and-restore.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (annabaker) [reply jekyll/_cci2/server-3-operator-backup-and-restore.adoc:394]: yeah, I saw that, that's why I was confused. I will add some clarifications and ping you on Slack if I have more questions, thank you!

File: jekyll/_cci2/server-3-operator-backup-and-restore.adoc (hunk 2 above)
Comment (nathanfishcircleci) [reply jekyll/_cci2/server-3-operator-backup-and-restore.adoc:415]: I would add something like "<YOUR_MINIO_ACCESS_KEY_ID>" with some context around, the minio access key to be used by Velero.

**Review by nathanfishcircleci — APPROVED** [review]
I responded to @nsmith5 comments but overall I think this is a good first iteration.

**Review by zawzawaung-prog — APPROVED** [review]

File: jekyll/_cci2/server-3-operator-backup-and-restore.adoc [hunk 3]
```diff
@@ -366,6 +370,120 @@ velero-68788b675c-dm2s7   1/1     Running   0          2m
...
```
Comment (rosieyohannan) [inline jekyll/_cci2/server-3-operator-backup-and-restore.adoc:415]:
```suggestion
mc admin user add $ALIAS <YOUR_MINIO_ACCESS_KEY_ID> <YOUR_MINIO_SECRET_ACCESS_KEY>
```

File: jekyll/_cci2/server-3-operator-backup-and-restore.adoc [hunk 4]
```diff
@@ -366,6 +370,120 @@ velero-68788b675c-dm2s7   1/1     Running   0          2m
...
+
+[source, bash]
+----
+mc mb ${ALIAS}/<YOUR_BUCKET>
+----
+
+=== Set 3 - Create a user and policy
+
+Next, create a user and policy for Velero to access your bucket.
+
```
Comment (rosieyohannan) [inline jekyll/_cci2/server-3-operator-backup-and-restore.adoc:411]:
```suggestion

NOTE: In the following snippet `<YOUR_MINIO_ACCESS_KEY_ID>` and `<YOUR_MINIO_SECRET_ACCESS_KEY>` refer to the credentials used by Velero to access MinIO.
```

File: jekyll/_cci2/server-3-operator-backup-and-restore.adoc (hunk 2 above)
Comment (rosieyohannan) [reply jekyll/_cci2/server-3-operator-backup-and-restore.adoc:415]: I've added suggestions to hopefully make things clearer on your suggestions!

**Review by nathanfishcircleci — APPROVED** [review]
I'd like to see a little of an overview on backup and restore. For example "Backup and restore of the CircleCI services is dependent on Velero. If your cluster is lost, you will not be able to restore CircleCI until you have successfully brought up Velero in the cluster. From there you can recover the CircleCI services." 

I think at the end we can probably include some additional details about making sure the configuration in KOTs admin uses the same IAM users (created when setting up velero) and same Buckets (created when setting up velero). I know it may seem obvious but we probably want to make sure folks tie the setup of velero. Maybe directly mentioning "CircleCI backup and restore now needs to be configured so it is aware of the Velero service you just finished creating".

**Comment by rosieyohannan** [comment]
> I'd like to see a little of an overview on backup and restore. For example "Backup and restore of the CircleCI services is dependent on Velero. If your cluster is lost, you will not be able to restore CircleCI until you have successfully brought up Velero in the cluster. From there you can recover the CircleCI services."
> 
> I think at the end we can probably include some additional details about making sure the configuration in KOTs admin uses the same IAM users (created when setting up velero) and same Buckets (created when setting up velero). I know it may seem obvious but we probably want to make sure folks tie the setup of velero. Maybe directly mentioning "CircleCI backup and restore now needs to be configured so it is aware of the Velero service you just finished creating".

@nathanfishcircleci  I'm coming in to this PR cold really but that notes about using the same IAM users and buckets sound like they should be at the top of the doc in prerequisites and recommendations maybe? Otherwise people could go through the guide not doing this and find out at the end? I'll stick them in anyway and see what you think.

**Comment by rosieyohannan** [comment]
@nathanfishcircleci I'm going to merge this into the 3.1 branch and we can continue working on it there if necessary

**Review by nathanfishcircleci — COMMENTED** [review]
This looks good to me.

**Pull request merged**